### ISSUE TITLE
Unclear error when no `parsedUrl` is provided

### ISSUE THREAD

**Issue by maxdeviant** [zeit/next.js#1279]
I have a project that was using custom routing via Express. I just upgraded to `next@^2.0.0-beta.32`, and now I get this error:

```
Error: Please provide a parsed url to `handle` as third parameter. See https://github.com/zeit/next.js#custom-server-and-routing for an example.
```

I followed the instructions and added the parsed URL as the third argument:

```ts
app.prepare()
  .then(() => {
    const server = express();
    server.get('*', (req, res) => {
      // Added the parsed URL as a third argument, as directed.
      handle(req, res, parse(req.url));
    });
    server.listen(port, host, err => {
      if (err) {
        throw err;
      }
      console.log(`Server listening on ${host}:${port}.`);
    })
  });
```

I dug into the code a bit and found the [source](https://github.com/zeit/next.js/blob/master/server/index.js#L40-L42) of the error:

```js
if (!parsedUrl.query) {
  throw new Error('Please provide a parsed url to `handle` as third parameter. See https://github.com/zeit/next.js#custom-server-and-routing for an example.')
}
```

After doing some more digging, I realized that you have to pass a second argument to `url.parse` to get it to parse the query parameters (otherwise it leaves it as `null`):

```js
> url.parse('https://localhost:3000')
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'localhost:3000',
  port: '3000',
  hostname: 'localhost',
  hash: null,
  search: null,
  query: null,
  pathname: '/',
  path: '/',
  href: 'https://localhost:3000/' }
> url.parse('https://localhost:3000', true)
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'localhost:3000',
  port: '3000',
  hostname: 'localhost',
  hash: null,
  search: '',
  query: {},
  pathname: '/',
  path: '/',
  href: 'https://localhost:3000/' }
```

**TL;DR:** I think the error that we are returning when no `parsedUrl` is passed is rather confusing, and should be improved.

**Comment by sedubois** [comment]
normally you can just `server.get('*', (req, res) => handle(req, res));`

**Comment by maxdeviant** [comment]
@sedubois I realize that now.

My issue was previously I was passing `handle` first-class:

```js
server.get('*', handle);
```

Which was passing the `next` argument to handle as the `parsedUrl`. This skipped the built-in URL parsing, and gave me that error.

Perhaps the error should specify that you can leave it off if you want the default behavior? Because without reading the source, no one would know that :smile:
